lock-api/src/test/java/com/palantir/lock/client/UnreliableTimelockClientTest.java

```diff
@@ -0,0 +1,137 @@
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public final class UnreliableTimelockClientTest {
+
+    private static final LockToken LOCK_TOKEN = LockToken.of(UUID.fromString("00000000-0000-0000-0000-000000000000"));
+    private static final LockDescriptor LOCK_DESCRIPTOR = StringLockDescriptor.of("foo");
+    private static final LockRequest LOCK_REQUEST = LockRequest.of(Set.of(LOCK_DESCRIPTOR), 1);
+
+    @Mock
+    private TimeLockClient timeLockClient;
+
+    private UnreliableTimeLockService unreliableTimeLockService;
+
+    @Before
+    public void before() {
+        unreliableTimeLockService = new UnreliableTimeLockService(timeLockClient, NoOpBuggifyFactory.INSTANCE);
+    }
+
+    @Test
+    public void verifyIsInitializedCallsDelegate() {
+        unreliableTimeLockService.isInitialized();
+        verify(timeLockClient).isInitialized();
+    }
+
+    @Test
+    public void verifyGetFreshTimestampCallsDelegate() {
+        unreliableTimeLockService.getFreshTimestamp();
+        verify(timeLockClient).getFreshTimestamp();
+    }
+
+    @Test
+    public void verifyGetCommitTimestampsCallsDelegate() {
+        unreliableTimeLockService.getCommitTimestamp(1, LOCK_TOKEN);
+        verify(timeLockClient).getCommitTimestamp(1, LOCK_TOKEN);
+    }
+
+    @Test
+    public void verifyGetFreshTimestampsCallsDelegate() {
+        unreliableTimeLockService.getFreshTimestamps(1);
+        verify(timeLockClient).getFreshTimestamps(1);
+    }
+
+    @Test
+    public void verifyLockImmutableTimestampCallsDelegate() {
+        unreliableTimeLockService.lockImmutableTimestamp();
+        verify(timeLockClient).lockImmutableTimestamp();
+    }
+
+    @Test
+    public void verifyStartIdentifiedAtlasDbTransactionCallsDelegate() {
+        unreliableTimeLockService.startIdentifiedAtlasDbTransactionBatch(1);
+        verify(timeLockClient).startIdentifiedAtlasDbTransactionBatch(1);
+    }
+
+    @Test
+    public void verifyGetImmutableTimestampCallsDelegate() {
+        unreliableTimeLockService.getImmutableTimestamp();
+        verify(timeLockClient).getImmutableTimestamp();
+    }
+
+    @Test
+    public void verifyLockCallsDelegate() {
+        unreliableTimeLockService.lock(LOCK_REQUEST);
+        verify(timeLockClient).lock(LOCK_REQUEST);
+    }
+
+    @Test
+    public void verifyLockWithOptionsCallsDelegate() {
+        unreliableTimeLockService.lock(LOCK_REQUEST, ClientLockingOptions.getDefault());
+        verify(timeLockClient).lock(LOCK_REQUEST, ClientLockingOptions.getDefault());
+    }
+
+    @Test
+    public void verifyWaitForLocksCallsDelegate() {
+        WaitForLocksRequest waitForLocksRequest = WaitForLocksRequest.of(Set.of(LOCK_DESCRIPTOR), 1);
+        unreliableTimeLockService.waitForLocks(waitForLocksRequest);
+        verify(timeLockClient).waitForLocks(waitForLocksRequest);
```

Comment:
This one works because we passed in a no-op buggify factory, though it seems like the logic that buggify does isn't tested (similarly, for lock/lock with options we ended up not testing the buggified behaviour). Should we have a version of this test that takes an always approach and check it behaves how we want it to?

Reply:
Yup yup, we should do that!